### PR TITLE
8273921: Refactor NSK/JDI tests to create thread using factory

### PR DESCRIPTION
**Pull request by lmesnik**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273921](https://bugs.openjdk.java.net/browse/JDK-8273921): Refactor NSK/JDI tests to create thread using factory

### Reviewers
 * [Chris Plummer](https://openjdk.java.net/census#cjplummer) (@plummercj - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)
 * [Alex Menkov](https://openjdk.java.net/census#amenkov) (@alexmenkov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5515/head:pull/5515` \
`$ git checkout pull/5515`

Update a local copy of the PR: \
`$ git checkout pull/5515` \
`$ git pull https://git.openjdk.java.net/jdk pull/5515/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5515`

View PR using the GUI difftool: \
`$ git pr show -t 5515`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5515.diff">https://git.openjdk.java.net/jdk/pull/5515.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIThreadFactory.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+public class JDIThreadFactory {
+
+    private static ThreadFactory threadFactory = r -> new Thread(r);
+
+    public static Thread newThread(JDITask task) {
+        Thread t = threadFactory.newThread(task);
+        t.setName(task.getName());
+        return t;
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIThreadFactory.java:39]: How about:
`    return newThread(task, task.getName());`

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/EventRequestManager/stepRequests/stepreq002a.java [hunk 2]
```diff
@@ -137,7 +137,7 @@ static int threadStart(Thread t) {
 
 }
 
-class Thread1stepreq002a extends Thread {
+class Thread1stepreq002a extends JDITask {
 
     String tName = null;
```
Comment (plummercj) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/EventRequestManager/stepRequests/stepreq002a.java:142]: Another case where tName was not removed.

**Review by plummercj — COMMENTED** [review]
I've gotten through about 1/3 of the test plus the new files, so thought I'd pass along my comments so far. Overall it looks good though.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ClassType/newInstance/newinstance001.java [hunk 3]
```diff
@@ -51,7 +51,7 @@
  * The test works as follows.                                   <BR>
  * A debugger launchs a debuggee which creates new thread, thread2. <BR>
  * The tested class TestClass is specified outside of the class <BR>
- *      Threadnewinstance002a extends Thread                    <BR>
+ *      Threadnewinstance002a extends JDITask                    <BR>
```
Comment (sspitsyn) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ClassType/newInstance/newinstance001.java:54]: Formatting nit: extra space.

**Review by sspitsyn — APPROVED** [review]
Hi Leonid,
The refactoring looks great.
I also like that some unneeded constructors and fields go away.
Thanks,
Serguei

**Comment by plummercj** [comment]
> I've gotten through about 1/3 of the test plus the new files, so thought I'd pass along my comments so far. Overall it looks good though.

I went through the rest of the files. No new issues, so just waiting for a response to my previous comments.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt001a.java [hunk 4]
```diff
@@ -162,7 +164,7 @@ static void m11() {
     }
 }
 
-class Thread1filter_rt001a extends Thread {
+class Thread1filter_rt001a extends JDITask {
 
     private String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt001a.java:169]: One more case where tName is not removed

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt001a.java [hunk 5]
```diff
@@ -198,7 +200,7 @@ static void m21() {
     }
 }
 
-class Thread2filter_rt001a extends Thread {
+class Thread2filter_rt001a extends JDITask {
 
     private String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt001a.java:205]: One more case where tName was not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt003a.java [hunk 6]
```diff
@@ -167,7 +170,7 @@ static void m11() {
     }
 }
 
-class Thread1filter_rt003a extends Thread {
+class Thread1filter_rt003a extends JDITask {
 
     private String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/StepRequest/addClassFilter_rt/filter_rt003a.java:175]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter001a.java [hunk 7]
```diff
@@ -138,7 +138,7 @@ static int threadStart(Thread t) {
     }
 
 
-    static class Thread2addthreadfilter001a extends Thread {
+    static class Thread2addthreadfilter001a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter001a.java:143]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter003a.java [hunk 8]
```diff
@@ -137,7 +137,7 @@ static int threadStart(Thread t) {
     static Object lockingObj[] = new Object[2];
     static volatile int number = 0;
 
-    static class Threadaddthreadfilter003a extends Thread {
+    static class Threadaddthreadfilter003a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter003a.java:142]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter004a.java [hunk 9]
```diff
@@ -171,7 +171,7 @@ static int threadStart(Thread t) {
     static Object lockingObj[] = new Object[2];
     static volatile int number = 0;
 
-    static class Threadaddthreadfilter004a extends Thread {
+    static class Threadaddthreadfilter004a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter004a.java:176]: One more case when tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter005a.java [hunk 10]
```diff
@@ -152,7 +152,7 @@ static int threadStart(Thread t) {
     static Object lockingObj[] = new Object[2];
     static volatile int number = 0;
 
-    static class Threadaddthreadfilter005a extends Thread {
+    static class Threadaddthreadfilter005a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/ThreadDeathRequest/addThreadFilter/addthreadfilter005a.java:157]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassExclusionFilter/filter001a.java [hunk 11]
```diff
@@ -156,7 +156,7 @@ static void run1(Thread t) {
     }
 
 
-    static class Thread1filter001a extends Thread {
+    static class Thread1filter001a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassExclusionFilter/filter001a.java:161]: One more case. tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassExclusionFilter/filter002a.java [hunk 12]
```diff
@@ -156,28 +156,25 @@ static void run1(Thread t) {
...
-            log1("  'run': exit   :: threadName == " + tName);
+            log1("  'run': exit   :: threadName == " + getName());
             return;
         }
     }
 
-    static class Thread2filter002a extends Thread {
+    static class Thread2filter002a extends JDITask {
 
         String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassExclusionFilter/filter002a.java:179]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt001a.java [hunk 13]
```diff
@@ -182,28 +182,25 @@ static void method () {
...
-        filter_rt001a.log1("  'run': exit   :: threadName == " + tName);
+        filter_rt001a.log1("  'run': exit   :: threadName == " + getName());
         return;
     }
 }
 
-class Thread2filter_rt001a extends Thread {
+class Thread2filter_rt001a extends JDITask {
 
     String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt001a.java:205]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt002a.java [hunk 14]
```diff
@@ -182,28 +182,25 @@ static void method () {
...
-        filter_rt002a.log1("  'run': exit   :: threadName == " + tName);
+        filter_rt002a.log1("  'run': exit   :: threadName == " + getName());
         return;
     }
 }
 
-class Thread2filter_rt002a extends Thread {
+class Thread2filter_rt002a extends JDITask {
 
     String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt002a.java:205]: tName is not removed.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt005a.java [hunk 15]
```diff
@@ -232,7 +229,7 @@ static void method () {
     }
 }
 
-class Thread2filter_rt005a extends Thread {
+class Thread2filter_rt005a extends JDITask {
 
     String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt005a.java:234]: The same

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt006a.java [hunk 16]
```diff
@@ -232,7 +229,7 @@ static void method () {
     }
 }
 
-class Thread2filter_rt006a extends Thread {
+class Thread2filter_rt006a extends JDITask {
 
     String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_rt/filter_rt006a.java:234]: One more case

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_s/filter_s001a.java [hunk 17]
```diff
@@ -221,7 +218,7 @@ static void method () {
     }
 }
 
-class Thread2filter_s001a extends Thread {
+class Thread2filter_s001a extends JDITask {
 
     String tName = null;
```
Comment (alexmenkov) [inline test/hotspot/jtreg/vmTestbase/nsk/jdi/WatchpointRequest/addClassFilter_s/filter_s001a.java:223]: tName is not removed.

**Comment by lmesnik** [comment]
@plummercj, @sspitsyn, @alexmenkov  Thank you for review, I found some more tests to update. Will update PR soon.

File: test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIThreadFactory.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDIThreadFactory.java]: fixed

File: test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDITask.java [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package nsk.share.jdi;
+
+public abstract class JDITask implements Runnable {
```
[the remark this answers is not in the document]
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/share/jdi/JDITask.java]: Renamed to NamedTask.

File: test/hotspot/jtreg/vmTestbase/nsk/jdi/BreakpointRequest/addInstanceFilter/instancefilter004a.java [hunk 19]
```diff
@@ -149,7 +149,7 @@ static int threadStart(Thread t) {
     static Object lockingObj[] = new Object[2];
     static volatile int number = 0;
 
-    static class Threadinstancefilter004a extends Thread {
+    static class Threadinstancefilter004a extends JDITask {
 
         String tName = null;
```
[the remark this answers is not in the document]
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/jdi/BreakpointRequest/addInstanceFilter/instancefilter004a.java]: Too late realized that it is redundant and forget to fix some tests.
Updated and checked that tName is used anymore.

**Comment by lmesnik** [comment]
The tName was removed and some other tests were updated to use NamedTask.

**Review by sspitsyn — APPROVED** [review]
I've reviewed all incremental updates.
Looks good to me.

**Review by alexmenkov — APPROVED** [review]

**Review by plummercj — APPROVED** [review]

**Comment by lmesnik** [comment]
@plummercj, @sspitsyn, @alexmenkov Thank you for the review.

**Pull request closed without merging**